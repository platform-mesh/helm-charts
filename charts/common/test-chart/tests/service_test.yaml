suite: deployment
templates:
  - service.yaml
values:
  - ../values.yaml
chart:
  namespace: default
  appVersion: 0.1.0
tests:
  - it: test default
    asserts:
      - matchSnapshot: {}
  - it: test extras
    asserts:
      - matchSnapshot: { }
    set:
      extraEnvs:
        - name: MY_ENV
          value: my-value
      extraServicePorts:
        - name: my-port
          port: 8081
          targetPort: 8081
  - it: test service type
    asserts:
      - matchSnapshot: { }
    set:
      service:
        type: NodePort