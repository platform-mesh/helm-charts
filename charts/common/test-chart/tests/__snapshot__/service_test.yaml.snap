test default:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME-test-chart
      type: ClusterIP
test extras:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
        - name: my-port
          port: 8081
          targetPort: 8081
      selector:
        app: RELEASE-NAME-test-chart
      type: ClusterIP
test service type:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      name: RELEASE-NAME-test-chart
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
      selector:
        app: RELEASE-NAME-test-chart
      type: NodePort
